### ISSUE TITLE
DateTime picker

### ISSUE THREAD

**Issue by gtaschuk** [networknt/react-schema-form#82]
We're trying to capture DateTimes with this form library and need a DateTime picker

@stevehu  let me know if this all makes sense/seems like the right type of widget for the library - if so, I'll put a bounty on it

## DateTime Picker
Timestamps should be captured in ISO 8601 format and UTC timezone by default, i.e.
`YYYY-MM- DDThh:mm:ss.sssZ`
If there is a local timezone offset used when capturing a time, the date should be recorded in UTC time
For example, if a user enters midnight New York (EST) time for March 15th 2018,
this should be saved as “2018-03- 14T20:00:00.000+00:00”

For the field:
◦ User should be able to click on material-ui icons next to the input and get
    ▪  the Date Picker to fill in the date portion
    ▪ The Time Picker to fill in the time portion
◦ User should be able to select the UTC offset of their time input
- So it might look like _________ [date picker button] __________[time picker button] < UTC offset select field >

**Comment by stevehu** [comment]
@gtaschuk You are right. We only have a DatePicker and default the time to midnight. The reason to use the UTC timezone is due to JSON convention. All JSON parser can recognize this format. If another format is passed to the backend, then the backend service needs to parse it based on special logic.

**Comment by gtaschuk** [comment]
Ok cool - I'm going to limit the spec above down to building a new datetime picker for the library (which I'll bounty), and we (TruSet) will figure out what we want to do for the date type

**Comment by stevehu** [comment]
the Date type is default to midnight and I don't think it will cause issues for the majority of the applications. Once the serialization is done, the backend application can decide to trim the time if desired. Another option is to use jdk8 module with Jackson but this ties the front end library with Java.

**Comment by gitcoinbot** [comment]
__This issue now has a funding of 0.06 ETH (22.53 USD @ $375.45/ETH) attached to it.__

 * If you would like to work on this issue you can claim it [here](https://gitcoin.co/issue/networknt/react-schema-form/82).
 * If you've completed this issue and want to claim the bounty you can do so [here](https://gitcoin.co/issue/networknt/react-schema-form/82)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $2141.22 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by gitcoinbot** [comment]
__Work has been started on the 0.06 ETH (23.11 USD @ $385.18/ETH) funding by__: 
 1. [@willsputra](https://gitcoin.co/profile/willsputra) 

 __Please work together__ and coordinate delivery of the issue scope. Gitcoin doesn't know enough about everyones skillsets / free time to say who should work on what, but we trust that the community is smart and well-intentioned enough to work together.  As a general rule; if you start work first, youll be at the top of the above list ^^, and should have 'dibs' as long as you follow through. 

 On the above list? Please leave a comment to let the funder  and the other parties involved what you're working, with respect to this issue and your plans to resolve it.  If you don't leave a comment, the funder may expire your submission at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/issue/networknt/react-schema-form/82)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $2676.63 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by gitcoinbot** [comment]
__Work has been started on the 0.06 ETH (23.11 USD @ $385.18/ETH) funding by__: 
 1. [@Toeplitz](https://gitcoin.co/profile/Toeplitz) 

 __Please work together__ and coordinate delivery of the issue scope. Gitcoin doesn't know enough about everyones skillsets / free time to say who should work on what, but we trust that the community is smart and well-intentioned enough to work together.  As a general rule; if you start work first, youll be at the top of the above list ^^, and should have 'dibs' as long as you follow through. 

 On the above list? Please leave a comment to let the funder  and the other parties involved what you're working, with respect to this issue and your plans to resolve it.  If you don't leave a comment, the funder may expire your submission at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/issue/networknt/react-schema-form/82)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $2561.07 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by vs77bb** [comment]
@toeplitz Any update here? Gitcoin Bot will remove you from this issue in two days if no progress is made.

**Comment by gitcoinbot** [comment]
__Work has been started on the 0.06 ETH (30.84 USD @ $513.94/ETH) funding by__: 
 1. [@ninjasun](https://gitcoin.co/profile/ninjasun) 

 __Please work together__ and coordinate delivery of the issue scope. Gitcoin doesn't know enough about everyones skillsets / free time to say who should work on what, but we trust that the community is smart and well-intentioned enough to work together.  As a general rule; if you start work first, youll be at the top of the above list ^^, and should have 'dibs' as long as you follow through. 

 On the above list? Please leave a comment to let the funder  and the other parties involved what you're working, with respect to this issue and your plans to resolve it.  If you don't leave a comment, the funder may expire your submission at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/issue/networknt/react-schema-form/82)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $3756.04 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started on the 0.06 ETH (38.88 USD @ $648.0/ETH) funding by__: 
 1. [@dimkk](https://gitcoin.co/profile/dimkk)  

 On the above list? Please leave a comment to let the funder  know your  plan. If you don't leave a comment, the funder may expire your submission at their discretion. 

 * Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/networknt/react-schema-form/82/199)
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $4,681.48 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
@dimkk are you still working on this issue?
* [x] warning 1 (3 days)
* [ ] warning 2 (6 days)
* [ ] auto removal (10 days)

**Comment by gitcoinbot** [comment]
@dimkk Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] auto removal (6 days)

**Comment by gitcoinbot** [comment]
@dimkk Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] auto removal (6 days)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.

 1. [@pfilippi24](https://gitcoin.co/profile/pfilippi24) 

 has committed to working on this project to be completed 3 weeks, 3 days ago.

**Comment by stevehu** [comment]
I am thinking to support both. I have created the react16 branch and still working on the details as material-ui v1 is not backward compatible.

**Comment by nemaniarjun** [comment]
I would like to take this up, if no one else is working on this. :)

**Comment by stevehu** [comment]
@nemaniarjun The latest code is in react16 branch. I have only the material-ui date object there but never got a chance to test it out. If you want to take it, please create new components based on https://material-ui-next.com/demos/pickers/

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.

 1. [@nemaniarjun](https://gitcoin.co/profile/nemaniarjun) 

 has committed to working on this project to be completed 4 weeks ago.


__Please answer following questions/comments__ :

I am first cleaning up a bit of the react16 branch, making sure the "Date" field works as expected.
The code on the branch is based on the Materal-ui not Material-ui-next, this breaks most fields and button.

I have already done some work to fix the example buttons and the Date Field, will post more updates by tomorrow.

**Comment by nemaniarjun** [comment]
why the 4 weeks ago? by the gitcoinbot?

**Comment by nemaniarjun** [comment]
why the `4 weeks ago`? by the @gitcoinbot?

**Comment by nemaniarjun** [comment]
@stevehu The code in `react16` branch uses material-ui-next, and also in the pickers link that you shared, the Date Picker returns a string instead of an Object. Is this the desired behavior?

**Comment by stevehu** [comment]
@nemaniarjun The react16 is a new branch and still working in progress. The form collects all fields and then send to the backend in a JSON format. In JSON, there is no date type but just a string. The date object in Javascript has a toJSON method that outputs something like "2012-04-23T18:25:43.511Z" and it is the standard these days. As long as we can convert to this format in the model, we are good. The other alternative solution is to convert to milliseconds since 1970 but backend system needs to know this fields is a date/time. Thanks.

**Comment by gitcoinbot** [comment]
@nemaniarjun Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=1) | [3 days](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=3) | [5 days](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=5) | [10 days](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=10) | [100 days](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@nemaniarjun due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/networknt/react-schema-form/82/199) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] warning (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=1) | [3 days](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=3) | [5 days](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=5) | [10 days](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=10) | [100 days](https://gitcoin.co/issue/networknt/react-schema-form/82/199?snooze=100)

**Comment by vs77bb** [comment]
@stevehu Bumping this issue in case it slipped your eye... @nemaniarjun looks to have submitted a PR here a bit ago.

Also snoozing Gitcoin Bot here for 100 days, appreciate your patience with the bot 😄

**Comment by weberV** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.06 ETH (41.53 USD @ $692.1/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](http://localhost:8000/issue/networknt/react-schema-form/82/199).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $2,520.69 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gtaschuk** [comment]
@vs77bb At this point it seems like nobody is sucessfully completing the project at this funding level.  Can you return the funds?

**Comment by ceresstation** [comment]
Hey @gtaschuk! It looks like @nemaniarjun submitted a PR above that was merged? Are there still some outstanding items on this issue?

**Comment by gtaschuk** [comment]
updated the library and got it now.  Sorry for confusion

**Comment by gtaschuk** [comment]
Project Approved!

**Comment by nemaniarjun** [comment]
I am afk right now. Will update gitcoin with the 'submission' soon.
